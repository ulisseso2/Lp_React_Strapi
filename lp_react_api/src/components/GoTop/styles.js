import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    background-color: red;
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    background-color: red;
  `}
`;
