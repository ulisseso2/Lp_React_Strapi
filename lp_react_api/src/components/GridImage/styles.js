import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css``}
`;

export const Title = styled.div`
  ${({ theme }) => css``}
`;
export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    align-items: center;
  `}
`;
export const GridImg = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    margin: 10px;
  `}
`;

export const Img = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    transition: all 300ms ease-in-out;
    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
