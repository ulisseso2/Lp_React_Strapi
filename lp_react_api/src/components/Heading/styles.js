import styled from 'styled-components';
import { css } from 'styled-components';

export const Title = styled.h1`
  ${({ theme, colorDark }) => css`
    color: ${colorDark ? theme.colors.hot : theme.colors.dark};
  `};
`;
// styled component do componente Heading
