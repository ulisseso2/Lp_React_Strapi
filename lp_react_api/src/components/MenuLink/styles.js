import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    font-size: ${theme.font.sizes.small};
    text-decoration: none;
    padding: ${theme.spacings.small};
    color: ${theme.colors.promaryColor};
    font-weight: 400;
    position: relative;
    &::after {
      content: '';
      position: absolute;
      bottom: 0;
      left: 50%;
      height: 0.2rem;
      width: 0px;
      background-color: ${theme.colors.secondaryColor};
      transition: all 300ms ease-in-out;
    }
    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
