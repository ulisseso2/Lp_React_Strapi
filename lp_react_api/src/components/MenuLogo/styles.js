import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: flex;
    text-decoration: none;
    color: inherit;
    align-items: center;
  `}
  img {
    max-height: 4rem;
  }
`;
