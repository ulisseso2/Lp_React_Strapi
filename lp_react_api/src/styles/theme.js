export const theme = {
  colors: {
    primary: '#457B9D',
    light: '#A8DADC',
    dark: '#1D3557',
    hot: '#E63946',
    white: '#f7fff7',
    black: '#343434',
  },
  fonts: {
    family: {
      main: "'Open Sans', sans-serif",
      title: "'Montserrat', sans-serif",
      body: "'Open Sans', sans-serif",
    },
  },
};
//tema padrão da aplicação usado em vários componentes
