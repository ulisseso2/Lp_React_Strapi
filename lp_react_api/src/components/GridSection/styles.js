import styled, { css } from 'styled-components';

import { Title as HeadingContainer } from '../Heading/styles';
import { Container as Textcomponent } from '../Textcomponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${Textcomponent} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    overflow: hidden;
    width: 100%;

    @media ${theme.font.media.ltmedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const GridElement = styled.div`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }
    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 6rem;
      font-weight: 900;
      left: -4rem;
      top: -1.5rem;
      transform: rotate(7deg);
    }

`;
