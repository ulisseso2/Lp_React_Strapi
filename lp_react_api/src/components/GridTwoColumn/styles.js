import styled, { css } from 'styled-components';

import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme, background }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};
    @media ${theme.font.media.ltmedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      margin-bottom: ${theme.spacings.xlarge};
      color: ${background ? theme.colors.white : theme.colors.primaryColor};
    }
  `}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css``}
`;
export const ImgContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    max-width: 500px;
  `}
`;
export const Image = styled.img`
  ${({ theme }) => css`
    max-width: 500px;
  `}
`;
