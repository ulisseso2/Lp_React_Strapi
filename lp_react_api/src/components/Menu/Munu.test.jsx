import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render.theme';
import { Menu } from '.';

import linksMock from '../NavLinks/mock';
import { theme } from '../../styles/theme';
const logoData = {
  text: 'Logo',
  link: '#target',
};
describe('<Menu />', () => {
  it('should render Logo and Main Menu Nav', () => {
    const { container } = renderTheme(
      <Menu links={linksMock} logoData={logoData} />,
    );
    expect(screen.getByRole('heading', { name: 'Logo' })).toBeInTheDocument();
    expect(
      screen.getByRole('navigation', { name: 'Main menu' }),
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render menu mobile and button for open and close menu', () => {
    const { container } = renderTheme(
      <Menu links={linksMock} logoData={logoData} />,
    );

    const button = screen.getByLabelText('Open/Close menu');
    const menuContainer = button.nextSibling;

    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();

    fireEvent.click(button);

    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();

    fireEvent.click(menuContainer);

    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
  });

  it('should not render links', () => {
    const { container } = renderTheme(<Menu logoData={logoData} />);
    expect(
      screen.queryByRole('navigation', { name: 'Main menu' }).firstChild,
    ).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
