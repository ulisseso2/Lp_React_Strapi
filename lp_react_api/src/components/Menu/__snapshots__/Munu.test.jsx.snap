// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should not render links 1`] = `
<div>
  <button
    aria-label="Open/Close menu"
    class="sc-eCYdqJ eWmrhx"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="StyledIconBase-ea9ulj-0 lbJwfL"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="sc-dkzDqf iJszVO"
  >
    <div
      class="sc-bczRLJ kHytwx"
    >
      <div
        class="sc-hKMtZM iknnuD"
      >
        <h1
          class="sc-gsnTZi gGAKok"
        >
          <a
            class="sc-iBkjds lbdrgb"
            href="#target"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="sc-jSMfEi imugUk"
        />
      </div>
    </div>
  </div>
   
</div>
`;

exports[`<Menu /> should render Logo and Main Menu Nav 1`] = `
<div>
  <button
    aria-label="Open/Close menu"
    class="sc-eCYdqJ eWmrhx"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="StyledIconBase-ea9ulj-0 lbJwfL"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="sc-dkzDqf iJszVO"
  >
    <div
      class="sc-bczRLJ kHytwx"
    >
      <div
        class="sc-hKMtZM iknnuD"
      >
        <h1
          class="sc-gsnTZi gGAKok"
        >
          <a
            class="sc-iBkjds lbdrgb"
            href="#target"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="sc-jSMfEi imugUk"
        >
          <a
            class="sc-gKXOVf haskHW"
            href="#target1"
            target="_self"
          >
            link 1
          </a>
          <a
            class="sc-gKXOVf haskHW"
            href="#target2"
            target="_self"
          >
            link 2
          </a>
          <a
            class="sc-gKXOVf haskHW"
            href="#target3"
            target="_self"
          >
            link 3
          </a>
          <a
            class="sc-gKXOVf haskHW"
            href="#target4"
            target="_self"
          >
            link 4
          </a>
          <a
            class="sc-gKXOVf haskHW"
            href="#target5"
            target="_self"
          >
            link 5
          </a>
        </nav>
      </div>
    </div>
  </div>
   
</div>
`;
